crates/jstzd/src/task/octez_client.rs

```diff
@@ -467,4 +491,27 @@ mod test {
         let res = Address::try_from(input_text.to_owned());
         assert!(res.is_err_and(|e| e.to_string().contains("Missing public key")));
     }
+
+    #[test]
+    fn extract_integer() {
+        let input = "30000 ꜩ";
+        let res = OctezClient::extract_digits(input);
+        assert!(res.is_ok_and(|balance| balance == 30000f64));
+        let input = "random text";
+        let res = OctezClient::extract_digits(input);
+        assert!(res.is_err_and(|e| e
+            .to_string()
+            .contains("Input string did not match the pattern")));
+    }
+    #[test]
+    fn extract_float() {
+        let input = "15.23453 ꜩ";
+        let res = OctezClient::extract_digits(input);
+        assert!(res.is_ok_and(|balance| balance == 15.23453));
+        let input = "random text";
```

Comment:
Let's remove this random text part since it's already part of the integer test case.